libvips/foreign/svgload.c

```diff
@@ -390,15 +456,10 @@ vips_foreign_load_svg_is_a_buffer( const void *buf, size_t len )
 		return( 0 );
 	for( i = 0; i < 24; i++ )
 		if( !isascii( str[i] ) )
-		return( 0 );
+			return( 0 );
 
 	for( i = 0; i < 200 && i < len - 5; i++ ) {
-		char txt[5];
-
-		/* 5, since we include the \0 at the end.
-		 */
-		vips_strncpy( txt, buf + i, 5 );
-		if( strcasecmp( txt, "<svg" ) == 0 )
+		if( strncmp( (const char *) str + i, "<svg", 4 ) == 0 )
```

Comment:
I think this should be strncasecmp(), shouldn't it? The old code did the copy because I didn't want to test for strncasecmp(), but maybe we can just assume all strings.h have this now.